### ISSUE TITLE
Tandem profile FR

### ISSUE THREAD

**Issue by poutnikl** [poutnikl/Trekking-Poutnik#19]
Suggested by @LeopoldKellers
[tandem related modification of template](https://github.com/LeopoldKellers/Brouter-profiles/raw/master/BikeProfiles/Trekking-dry.brf)  , along with the [ final profile ](https://github.com/LeopoldKellers/Brouter-profiles/raw/master/BikeProfiles/Trekking-Tandem-dry.brf)  
in   [https://github.com/LeopoldKellers/Brouter-profiles](https://github.com/LeopoldKellers/Brouter-profiles)

See also  [BRouter Google group tandem profile topic](https://groups.google.com/forum/#!topic/osm-android-bikerouting/yr2ZfLcwGD4)

**Comment by poutnikl** [comment]
this dismountpenalty is applied twice if there is tandem specific obstacle and no bike general access.

should be rather
add ( if ( and barrier=cycle_barrier|chicane not allow_cycle_barrier ) then 
   ( add ( switch bikeaccess dismountpenalty 0 )   haulpenalty )

```
assign initialcost =
        add ( if ( and barrier=cycle_barrier|chicane not allow_cycle_barrier ) then ( add **dismountpenalty** haulpenalty )
              ......
              else initial_cost_node )
            if bikeaccess then 0  else ( if footaccess then **dismountpenalty** else 1000000 )
```

**Comment by poutnikl** [comment]
For evidence, quoted the original relevant part of the post by @LeopoldKellers in the above mentioned group.

The changes are the following:
- I replaced a hardcoded constant of value 4 into a parameter called
pushpenalty, as is believe that is essentially what it represents.
- Further, there was another hardcoded constant, which I wanted to make
adjustable by turning it into a variable with the name dismountpenalty.
The reason is that two people getting off a tandem requires
significantly more time than jumping off a single (possibly while still
rolling slowly).
- Finally, I added the name haulpenalty. The idea is to penalize the
effect of lifting the bike up and carrying it around by foot. Its
default value of 40 arises from preserving the hardcoded constant in the
check for stairs, where I applied this new parameter as well.
- When changing the new allow_cycle_barrier parameter from its default
value true to false, then an additional check  adds dismountpenalty and
haulpenalty at cycle barriers (somtimes also tagged as chicane).

Summing up, none of these changes have any effect on your profile unless
one of the newly defined parameters' defaults is altered by the user.
The tandem profile is then derived from that modified template by
tweaking the new parameters described above. Probably there are still
some details left to be improved, but quickly comparing the results in
brouter-web to some of my previous routes makes me confident that I did
not screw up in a major way at least.

I believe that my profile could also be beneficial to not only tandem
teams, but for example also people riding trikes. Or to those who have
physical issues restricting their ability to lift their own vehicle over
obstacles or carry it on stairs.

**Comment by LeopoldKellers** [comment]
> this dismountpenalty is applied twice if there is tandem specific obstacle and no bike general access.

You are totally right, somehow I only had situations like this in mind (as these are the most annoying ones): 
[http://pettmansuelm.de/dateien/Nachrichteninhalte/Ausgabe_3_2011/pms_2011-3_Umlaufsperren_06.jpg](url)

I fixed it and created the pull request. Around the weekend I will install windows into a virtual machine (need to do that anyway for a different reason) and see whether I can adapt your batch script to generate the profile from an updated template.

**Comment by poutnikl** [comment]
I guess the needed batch tweaking is just minor. Mainly 2 points:
   * Specifying paths to cygwin binaries wget and sed
   * Specifying location of the template.  For now it is hardcoded as being particular files in master branches in the Hiking/Car-Profile/Trekking-Poutnik repositores. See the batch for CLI parameters. "sedbatch bike" will generate only bike profiles. ( I have not used it for a while, so I may omit something)

For now, one cannot specify to generate just a particular profile( _Edit: In fact, it is easier to modify few values of the template and save it as the final profile_), but I may change it in future.

**Edit: It is quite possible the port to Linux shell scripts may not be that difficult for you, and in fact, it would be an appreciated contribution, as I am not good in Unix/Linux scripting.
Note that the profiles keep Unixlike LF EOLs and my Windows GIT automatically converts them to local CRLF and back.  It may creat EOL mismatch in windows if file context is obtains in different ways, so EOL would not match.**

**Comment by poutnikl** [comment]
Hi @LeopoldKellers ,  you can test the new template in the [tandem branch](https://github.com/poutnikl/Trekking-Poutnik/tree/tandem).   I have made few formal rearrangements, mainly putting the constants to the internal section of Global declaration. I want to keep the initial user tweaking part short for most used tweaking.

You may noticed few special profiles in the tandem and master branch. Originally, I kept special profiles in their respective template repository, instead of the profile storage repository. They are not in the develop branch anymore, as implemented into the template and became regulars.

Currently, I am not decided where  to keep special profiles, and I may not be consistent in it.

**Comment by LeopoldKellers** [comment]
> new template in the tandem branch. I have made few formal rearrangements, mainly putting the constants to the internal section of Global declaration.

Looks good :-) I reconsidered the name "allow_cycle_barriers". Instead, "ignore_cycle_barriers" seems to describe it better. I added a pull request.

I will look into modifying the batch file next, so that there is no need to worry about it as a special profile. I'll just adapt what you did there.
Personally I don't urgently need a unix shell script, but I'll keep the idea in the back of my mind for some cold, dark winter nights

**Comment by poutnikl** [comment]
I am wondering, if the better syntax would be  to have ` assign tandem_bike 0` and to define conditional expression for related global variables. Like `assign dismountpenalty  switch tandem_bike X  Y`.

I prefer the approach to have the switches to be rather high level ones, and to derive the low level effects
in the internal global context and/or way/node context. As the global context is evaluated once, one can make as complex expressions as needed,

**Comment by LeopoldKellers** [comment]
I agree. Did it already. Will also adapt the pull-request for automatic profile generation accordingly.

**Comment by poutnikl** [comment]
@LeopoldKellers Tandem feature merged to the Trekking-Poutnik.brf template, ( in both major develop and master branches )
